src/java/org/apache/cassandra/utils/FBUtilities.java

```diff
@@ -486,6 +486,81 @@ public static int nowInSeconds()
             Uninterruptibles.sleepUninterruptibly(delay, TimeUnit.MILLISECONDS);
         }
     }
+
+    /**
+     * Returns a new {@link Future} wrapping the given list of futures and returning a list of their results.
+     */
+    public static Future<List> allOf(Collection<Future> futures)
+    {
+        if (futures.isEmpty())
+            return CompletableFuture.completedFuture(null);
+
+        return new Future<List>()
+        {
+            @Override
+            @SuppressWarnings("unchecked")
+            public List get() throws InterruptedException, ExecutionException
+            {
+                List result = new LinkedList<>();
```

Comment:
Is there a reason for choosing a `LinkedList` over an `ArrayList` ?

Reply:
changed to `new ArrayList<>(futures.size());`